src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "HwndTerminal.hpp"
+#include <DefaultSettings.h>
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/dx/DxRenderer.hpp"
+#include "../../cascadia/TerminalCore/Terminal.hpp"
+#include "../../types/viewport.cpp"
+#include "../../types/inc/GlyphWidth.hpp"
+
+using namespace ::Microsoft::Terminal::Core;
+
+static LPCWSTR term_window_class = L"HwndTerminalClass";
+
+LRESULT CALLBACK HwndTerminalWndProc(
+    HWND hwnd,
+    UINT uMsg,
+    WPARAM wParam,
+    LPARAM lParam) noexcept
+{
+    return DefWindowProc(hwnd, uMsg, wParam, lParam);
+}
+
+bool RegisterTermClass(HINSTANCE hInstance) noexcept
+{
+    WNDCLASSW wc;
+    if (GetClassInfo(hInstance, term_window_class, &wc))
+    {
+        return true;
+    }
+
+    wc.style = 0;
+    wc.lpfnWndProc = HwndTerminalWndProc;
+    wc.cbClsExtra = 0;
+    wc.cbWndExtra = 0;
+    wc.hInstance = hInstance;
+    wc.hIcon = nullptr;
+    wc.hCursor = LoadCursor(nullptr, IDC_ARROW);
+    wc.hbrBackground = nullptr;
+    wc.lpszMenuName = nullptr;
+    wc.lpszClassName = term_window_class;
+
+    return RegisterClassW(&wc) != 0;
+}
+
+HwndTerminal::HwndTerminal(HWND parentHwnd) :
+    _desiredFont{ DEFAULT_FONT_FACE.c_str(), 0, 10, { 0, 14 }, CP_UTF8 },
+    _actualFont{ DEFAULT_FONT_FACE.c_str(), 0, 10, { 0, 14 }, CP_UTF8, false }
+{
+    HINSTANCE hInstance = wil::GetModuleInstanceHandle();
+
+    if (RegisterTermClass(hInstance))
+    {
+        _hwnd = wil::unique_hwnd(CreateWindowExW(
+            0,
+            term_window_class,
+            nullptr,
+            WS_CHILD |
+                WS_CLIPCHILDREN |

```

Comment:
nit: these are indented more than the surrounding?

Reply:
This is what `Invoke-CodeFormat` gives me.